File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

public class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        Map<Character, Integer> charToMinIndex = new HashMap<>();
        for (String line : keymap) {
            for (int idx = 0; idx < line.length(); idx++) {
                char c = line.charAt(idx);
                int position = idx + 1;
                charToMinIndex.put(c, Math.min(charToMinIndex.getOrDefault(c, Integer.MAX_VALUE), position));
            }
        }

        int[] result = new int[targets.length];
        for (int i = 0; i < targets.length; i++) {
            String target = targets[i];
            int cur = 0;
            for (char word : target.toCharArray()) {
                if (!charToMinIndex.containsKey(word)) {
                    cur = -1;
                    break;
                }
                cur += charToMinIndex.get(word);
            }
            result[i] = cur;
        }

        return result;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    char_to_min_index = {}
    for line in keymap:
        for idx, char in enumerate(line, start=1):
            if char not in char_to_min_index:
                char_to_min_index[char] = idx
            else:
                char_to_min_index[char] = min(char_to_min_index[char], idx)

    result = []
    for target in targets:
        cur = 0
        for word in target:
            if word not in char_to_min_index:
                cur = -1
                break
            cur += char_to_min_index[word]
        result.append(cur)
    
    return result